pkg/apis/componentconfig/v1alpha1/zz_generated.conversion.go

```diff
@@ -814,6 +814,7 @@ func autoConvert_v1alpha1_LeaderElectionConfiguration_To_componentconfig_LeaderE
 	out.LeaseDuration = in.LeaseDuration
 	out.RenewDeadline = in.RenewDeadline
 	out.RetryPeriod = in.RetryPeriod
+	out.LockType = in.LockType
```

Comment:
`endpoints` is a resource (addressable via the API) and is what most external components should use.  Kind (`Endpoints`) is a schema, and multiple resources can have that schema.